core/src/mfd.rs

```diff
@@ -0,0 +1,74 @@
+//! Implementation of a memory fd
+
+use std::io::{Read, Seek, SeekFrom, Write};
+use std::os::fd::{AsRawFd, RawFd};
+
+use anyhow::{bail, Result};
+use memfd::{FileSeal, Memfd, MemfdOptions};
+
+pub struct Mfd(Memfd);
+
+impl Mfd {
+    /// Creates an empty named Memfd
+    pub fn create(name: &str) -> Result<Self> {
+        let opts = MemfdOptions::default().allow_sealing(true);
+        let mfd = opts.create(name)?;
+        Ok(Self(mfd))
+    }
+
+    /// Reads all data available
+    pub fn read_all(&self) -> Result<Vec<u8>> {
+        self.0.as_file().seek(SeekFrom::Start(0))?;
+        let mut buf: Vec<u8> = Vec::with_capacity(128);
```

Comment:
No, it is just a pre-allocations, not a limit. The data in the memfd's is unlikely to exceed 128 byte. However, I think it might be a good idea to change it to `Vec::new()` instead.

Reply:
Maybe add a todo for the future: IMHO when the data is < ~8 pages, it makes more sense to not allocate a memfd but rather to just inline the data into the Socket Datagram. This would provide the best behavior for all kinds of messages.
On my system, `cat /proc/sys/net/core/wmem_max` reveals about 200 kBytes of is the maximum datagram size (`212992`).
`getconf PAGE_SIZE` asserts a 4KiByte page size, thus a reasonable threshold could be in the ballpark of 32KiB. Below that, inline the data, above that, create a memfd.

However, this adds extra complications, which requires more testing for good coverage. So please just add a TODO akin

> TODO evaluate whether inlining of small message directly to the datagram makes sense.

The preallocation is ok, but not needed IMHO.